src/messaging/ReliableMessageContext.cpp

```diff
@@ -296,6 +305,7 @@ void ReliableMessageContext::SetPendingPeerAckMessageCounter(uint32_t aPeerAckMe
 {
     mPendingPeerAckMessageCounter = aPeerAckMessageCounter;
     SetAckPending(true);
+    mFlags.Set(Flags::kFlagHasHadAckPending);
```

Comment:
"HasHad" in the name makes it hard to parse. I was at first wondering if SetAckPending(true) should have done it, but the answer is probably no. I am struggling, looking at the names of the methods and the flow, to really figure-out if anything is missing or going awry. There is a bit of state combinatorial explosion with different cases... Could `kFlagHasHadAckPending` be renamed to something grammatically less complex?

Reply:
How about `kFlagAckMessageCounterIsValid` ?  @tcarmelveilleux